### PR TITLE
delete badge prints endpoint + audit log model refactor

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3399168&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Audit/ConcreteFormatters/ChildEntityFormatters/SummitAttendeeBadgePrintAuditLogFormatter.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if ($child_entity_action_type == IChildEntityAuditLogFormatter::CHILD_ENTITY_DELETION &&
+            $subject instanceof SummitAttendeeBadgePrint) {
+            $print_date = $subject->getPrintDate()->format('Y-m-d H:i:s');
+            $view_type_name = 'N/A';
+            $view_type = $subject->getViewType();
+            if (!is_null($view_type)) {
+                $view_type_name = $view_type->getName();
+            }
+            $requestor = $subject->getRequestor();
+            return "SummitAttendeeBadgePrint with ID {$subject->getID()}, printed on {$print_date}, view type {$view_type_name} and requestor member id {$requestor->getId()} was removed";
```
Comment (smarcet) [inline app/Audit/ConcreteFormatters/ChildEntityFormatters/SummitAttendeeBadgePrintAuditLogFormatter.php:39]: we need to update the format
requestor member <FullName>( email)

File: app/Models/Foundation/Main/Audit/AuditLog.php [hunk 2]
```diff
@@ -23,6 +23,7 @@
  * @ORM\DiscriminatorMap({
  *     "SummitAuditLog" = "SummitAuditLog",
  *     "SummitEventAuditLog" = "SummitEventAuditLog",
+ *     "SummitAttendeeBadgeAuditLog" = "SummitAttendeeBadgeAuditLog"
```
Comment (smarcet) [inline app/Models/Foundation/Main/Audit/AuditLog.php:26]: you need to add a migration to update the Enumerate

Comment (romanetar) [reply app/Models/Foundation/Main/Audit/AuditLog.php:26]: done

File: app/Audit/ConcreteFormatters/ChildEntityFormatters/SummitAttendeeBadgePrintAuditLogFormatter.php (hunk 1 above)
Comment (romanetar) [reply app/Audit/ConcreteFormatters/ChildEntityFormatters/SummitAttendeeBadgePrintAuditLogFormatter.php]: done

File: app/Services/Model/Imp/SummitAttendeeBadgePrintService.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteBadgePrintsByTicket(Summit $summit, int $ticket_id): void
+    {
+        $this->tx_service->transaction(function() use($summit, $ticket_id){
+
+            $ticket = $this->ticket_repository->getById($ticket_id);
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitAttendeeBadgePrintService.php:60]: add debug log here so we could keep track that is being called

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitAttendeeBadgePrintApiController.php [hunk 4]
```diff
@@ -147,4 +154,21 @@ function () {
...
+     * @param $ticket_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function deleteBadgePrints($summit_id, $ticket_id)
+    {
+        return $this->processRequest(function () use ($summit_id, $ticket_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->getResourceServerContext())->find(intval($summit_id));
+            if (is_null($summit)) return $this->error404();
+
+            $this->service->deleteBadgePrintsByTicket($summit, $ticket_id);
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitAttendeeBadgePrintApiController.php:169]: convert ticket id to int using intval

File: app/Services/Model/Imp/SummitAttendeeBadgePrintService.php (hunk 3 above)
Comment (romanetar) [reply app/Services/Model/Imp/SummitAttendeeBadgePrintService.php:60]: done

File: app/Services/Model/Imp/SummitAttendeeBadgePrintService.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+            if (!$ticket instanceof SummitAttendeeTicket)
+                throw new EntityNotFoundException(sprintf("ticket id %s not found", $ticket_id));
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Services/Model/Imp/SummitAttendeeBadgePrintService.php:63]: done

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitAttendeeBadgePrintApiController.php (hunk 4 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitAttendeeBadgePrintApiController.php:169]: done

File: app/Services/Model/Imp/SummitAttendeeBadgePrintService.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+            Log::debug(sprintf("SummitAttendeeBadgePrintService::deleteBadgePrintsByTicket: summit id %s, ticket id %s", $summit->getId(), $ticket_id));
+
+            $ticket = $this->ticket_repository->getById($ticket_id);
+
+            if (!$ticket instanceof SummitAttendeeTicket)
+                throw new EntityNotFoundException(sprintf("Ticket id %s not found.", $ticket_id));
+
+            $badge = $ticket->getBadge();
+            if (!is_null($badge)) $badge->clearPrints();
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitAttendeeBadgePrintService.php:69]: this is not working

Comment (smarcet) [reply app/Services/Model/Imp/SummitAttendeeBadgePrintService.php:69]: we need to store prints on another backup table before delete it
its required on the ticket

File: app/Services/Model/Imp/SummitAttendeeBadgePrintService.php [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * @inheritDoc
+     */
+    public function deleteBadgePrintsByTicket(Summit $summit, int $ticket_id): void
+    {
+        $this->tx_service->transaction(function() use($summit, $ticket_id){
+
+            Log::debug(sprintf("SummitAttendeeBadgePrintService::deleteBadgePrintsByTicket: summit id %s, ticket id %s", $summit->getId(), $ticket_id));
+
+            $ticket = $this->ticket_repository->getById($ticket_id);
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitAttendeeBadgePrintService.php:63]: u need to check that ticket belongs to this summit ( check the order )

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**